sources/OpenMcdf/CFItem.cs

```diff
@@ -12,260 +11,204 @@
 namespace OpenMcdf
 {
     /// <summary>
-    /// Abstract base class for Structured Storage entities.
+    ///     Abstract base class for Structured Storage entities.
     /// </summary>
     /// <example>
-    /// <code>
+    ///     <code>
+    ///  
+    ///  const String STORAGE_NAME = "report.xls";
+    ///  CompoundFile cf = new CompoundFile(STORAGE_NAME);
     /// 
-    /// const String STORAGE_NAME = "report.xls";
-    /// CompoundFile cf = new CompoundFile(STORAGE_NAME);
-    ///
-    /// FileStream output = new FileStream("LogEntries.txt", FileMode.Create);
-    /// TextWriter tw = new StreamWriter(output);
-    ///
-    /// // CFItem represents both storage and stream items
-    /// VisitedEntryAction va = delegate(CFItem item)
-    /// {
-    ///      tw.WriteLine(item.Name);
-    /// };
-    ///
-    /// cf.RootStorage.VisitEntries(va, true);
-    ///
-    /// tw.Close();
+    ///  FileStream output = new FileStream("LogEntries.txt", FileMode.Create);
+    ///  TextWriter tw = new StreamWriter(output);
     /// 
-    /// </code>
+    ///  // CFItem represents both storage and stream items
+    ///  VisitedEntryAction va = delegate(CFItem item)
+    ///  {
+    ///       tw.WriteLine(item.Name);
+    ///  };
+    /// 
+    ///  cf.RootStorage.VisitEntries(va, true);
+    /// 
+    ///  tw.Close();
+    ///  
+    ///  </code>
     /// </example>
-    public abstract class CFItem : IComparable<CFItem>
+    public abstract class CFItem : IComparable<CFItem>, IEquatable<CFItem>
     {
-        private CompoundFile compoundFile;
-
-        protected CompoundFile CompoundFile
-        {
-            get { return compoundFile; }
-        }
-
-        protected void CheckDisposed()
-        {
-            if (compoundFile.IsClosed)
-                throw new CFDisposedException("Owner Compound file has been closed and owned items have been invalidated");
-        }
-
-        protected CFItem()
-        {
-        }
-
         protected CFItem(CompoundFile compoundFile)
         {
-            this.compoundFile = compoundFile;
-        }
-
-        #region IDirectoryEntry Members
-
-        private IDirectoryEntry dirEntry;
-
-        internal IDirectoryEntry DirEntry
-        {
-            get { return dirEntry; }
-            set { dirEntry = value; }
-        }
-
-
-
-        internal int CompareTo(CFItem other)
-        {
-
-            return this.dirEntry.CompareTo(other.DirEntry);

```

Comment:
In General, this seems to be incorrect as we ommit the second property CompoundFile - but i am not sure if the current implementation relies on that in any way, therefore i left it hanging like that.